godel_process_planning/src/common_utils.cpp

```diff
@@ -0,0 +1,257 @@
+                                                    trajectory_msgs::JointTrajectory& traj)
+{
+  traj.joint_names = joints;
+  traj.header.frame_id = DEFAULT_FRAME_ID;
+  traj.header.stamp = ros::Time::now();
+}
+
+std::vector<double> godel_process_planning::getCurrentJointState(const std::string& topic)
+{
+  sensor_msgs::JointStateConstPtr state = ros::topic::waitForMessage<sensor_msgs::JointState>(topic, ros::Duration(0.0));
+  if (!state) throw std::runtime_error("Joint state message capture failed");
+  return state->position;
+}
+
+godel_process_planning::DescartesTraj
+godel_process_planning::createLinearPath(const Eigen::Affine3d &start,
+                                         const Eigen::Affine3d &stop,
+                                         double ds)
+{
+  using namespace descartes_trajectory;
+
+  PoseVector cart_path = interpolateCartesian(start, stop, ds);
+  DescartesTraj result;
+  for (std::size_t i = 0; i < cart_path.size(); ++i)
+  {
+    result.push_back(boost::make_shared<AxialSymmetricPt>(cart_path[i],
+                                                          DEFAULT_ANGLE_DISCRETIZATION,
+                                                          descartes_trajectory::AxialSymmetricPt::Z_AXIS));
+  }
+
+  return result;
+}
+
+
+godel_process_planning::DescartesTraj
+godel_process_planning::createJointPath(const std::vector<double> &start,
+                                        const std::vector<double> &stop,
+                                        double dtheta)
+{
+  JointVector path = interpolateJoint(start, stop, dtheta);
+  DescartesTraj result;
+  for (std::size_t i = 0; i < path.size(); ++i)
+  {
+    result.push_back(boost::make_shared<descartes_trajectory::JointTrajectoryPt>(path[i]));
+  }
+  return result;
+}
+
+
+trajectory_msgs::JointTrajectory godel_process_planning::getMoveitPlan(const std::string &group_name,
+                                                                       const std::vector<double> &joints_start,
+                                                                       const std::vector<double> &joints_stop,
+                                                                       moveit::core::RobotModelConstPtr model)
+{
+  const moveit::core::JointModelGroup* group = model->getJointModelGroup(group_name);
+  robot_state::RobotState goal_state(model);
+  goal_state.setJointGroupPositions(group, joints_stop);
+
+  moveit_msgs::GetMotionPlan::Request req;
+  req.motion_plan_request.group_name = group_name;
+  req.motion_plan_request.num_planning_attempts = DEFAULT_MOVEIT_NUM_PLANNING_ATTEMPTS;
+  req.motion_plan_request.max_velocity_scaling_factor = DEFAULT_MOVEIT_VELOCITY_SCALING;
+  req.motion_plan_request.allowed_planning_time = DEFAULT_MOVEIT_PLANNING_TIME;
+  req.motion_plan_request.planner_id = DEFAULT_MOVEIT_PLANNER_ID;
+
+  req.motion_plan_request.workspace_parameters.header.frame_id = DEFAULT_MOVEIT_FRAME_ID;
```

Comment:
The value for the frame_id can be obtained directly from the moveit Robot Model.